### PR TITLE
perf(math): optimize `LagrangeInterpolate`

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR optimizes `LagrangeInterpolate()` by parallelizing the outer loop instead of the inner loop.

### CODE REVIEW

File: tachyon/math/polynomials/univariate/lagrange_interpolation.h [hunk 1]
```diff
@@ -52,36 +52,58 @@ bool LagrangeInterpolate(
...
-      }
-      ++k_start;
-      coeffs[0] *= -points[j];
-    }
+  std::vector<std::vector<F>> coeffs_sums = base::ParallelizeMap(
+      evals, [&points, &denoms](absl::Span<const F> chunk, size_t chunk_offset,
+                                size_t chunk_size) {
+        // See comments in |UnivariateDenseCoefficients::FromRoots()|.
+        // clang-format off
+        // NOTE(chokobole): This computes the polynomial whose roots are {x₀, ..., xᵢ₋₁, xᵢ₊₁, ..., xₙ₋₁}.
```
Comment (ashjeong) [inline tachyon/math/polynomials/univariate/lagrange_interpolation.h:60]: does it compute xᵢ?

Comment (chokobole) [reply tachyon/math/polynomials/univariate/lagrange_interpolation.h:60]: Nope. it skips. Please check the line 83.

```c++
if (i == j) continue;
```

File: tachyon/math/polynomials/univariate/lagrange_interpolation.h [hunk 2]
```diff
@@ -52,36 +52,58 @@ bool LagrangeInterpolate(
...
+        // Nᵢ(X) = (X - x₀)...(X - xᵢ₋₁)(X - xᵢ₊₁)...(X - xₙ₋₁)
+        //       = cₙ₋₁Xⁿ⁻¹ + cₙ₋₂Xⁿ⁻² + ... + c₁X¹ + c₀X⁰
+        // |coeffs[i]| = cᵢ
+        // clang-format on
+
+        std::vector<F> coeffs = base::CreateVector(points.size(), F::Zero());
+        std::vector<F> coeffs_sum =
+            base::CreateVector(points.size(), F::Zero());
```
Comment (TomTaehoonKim) [inline tachyon/math/polynomials/univariate/lagrange_interpolation.h:68]:
```suggestion
        std::vector<F> coeffs(points.size());
        std::vector<F> coeffs_sum(points.size());
```

**Review by TomTaehoonKim — CHANGES_REQUESTED** [review]

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**